File: app/src/main/java/com/sample/instagramlikecolortransition/MainActivity.java
/*
 * Copyright (c) 2018, Deepak and/or its affiliates. All rights reserved.
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * - Redistributions of source code must retain the above copyright
 *   notice, this list of conditions and the following disclaimer.
 *
 * - Redistributions in binary form must reproduce the above copyright
 *   notice, this list of conditions and the following disclaimer in the
 *   documentation and/or other materials provided with the distribution.
 */

package com.sample.instagramlikecolortransition;

import android.graphics.drawable.AnimationDrawable;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {
    private AnimationDrawable anim;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // reference to parent layout
        LinearLayout container = findViewById(R.id.container);

        // initialize the animation
        anim = (AnimationDrawable) container.getBackground();
        anim.setEnterFadeDuration(6000);
        anim.setExitFadeDuration(2000);
    }

    @Override
    protected void onResume() {
        super.onResume();
        // start the animation
        if (anim != null && !anim.isRunning())
            anim.start();
    }

    @Override
    protected void onPause() {
        super.onPause();
        // stop the animation
        if (anim != null && anim.isRunning())
            anim.stop();
    }
}

File: kotlin/src/main/java/com/sample/instagramlikecolortransition/kotlin/MainActivity.kt
package com.sample.instagramlikecolortransition.kotlin

import android.graphics.drawable.AnimationDrawable
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.LinearLayout

class MainActivity : AppCompatActivity() {
    private lateinit var anim: AnimationDrawable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // reference to parent layout
        val container = findViewById<LinearLayout>(R.id.container)

        // initialize the animation
        anim = container.background as AnimationDrawable
        anim?.setEnterFadeDuration(6000)
        anim?.setExitFadeDuration(2000)
    }

    override fun onResume() {
        super.onResume()
        // start the animation
        if (!anim?.isRunning())
            anim.start()
    }

    override fun onPause() {
        super.onPause()
        // stop the animation
        if (anim?.isRunning())
            anim.stop()
    }

}
